src/OracleService/OracleService.cs

```diff
@@ -314,18 +315,23 @@ private async Task<(OracleResponseCode, string)> ProcessUrlAsync(string url)
             }
         }
 
-        public static Transaction CreateResponseTx(DataCache snapshot, OracleRequest request, OracleResponse response, ECPoint[] oracleNodes, ProtocolSettings settings)
+        public static Transaction CreateResponseTx(DataCache snapshot, OracleRequest request, OracleResponse response, ECPoint[] oracleNodes, ProtocolSettings settings, bool useCurrentHeight = false)
         {
             var requestTx = NativeContract.Ledger.GetTransactionState(snapshot, request.OriginalTxid);
             var n = oracleNodes.Length;
             var m = n - (n - 1) / 3;
             var oracleSignContract = Contract.CreateMultiSigContract(m, oracleNodes);
-
+            uint height = NativeContract.Ledger.CurrentIndex(snapshot);
+            var validUntilBlock = requestTx.BlockIndex + settings.MaxValidUntilBlockIncrement;
+            while (useCurrentHeight && validUntilBlock <= height)
+            {
+                validUntilBlock += settings.MaxValidUntilBlockIncrement;
```

Comment:
We could check that `MaxValidUntilBlockIncrement` is not 0, but "is not" possible...

```csharp
public uint MaxValidUntilBlockIncrement => 86400000 / MillisecondsPerBlock;
```